users/src/main/java/io/crate/operation/user/UserManagerService.java

```diff
@@ -0,0 +1,183 @@
+                              ClusterService clusterService) {
+        this.transportCreateUserAction = transportCreateUserAction;
+        this.transportDropUserAction = transportDropUserAction;
+        clusterService.add(this);
+    }
+
+    static Set<User> getUsersFromMetaData(@Nullable UsersMetaData metaData) {
+        ImmutableSet.Builder<User> usersBuilder = new ImmutableSet.Builder<User>().add(CRATE_USER);
+        if (metaData != null) {
+            for (String userName : metaData.users()) {
+                usersBuilder.add(new User(userName, false));
+            }
+        }
+        return usersBuilder.build();
+    }
+
+    @Override
+    public CompletableFuture<Long> createUser(CreateUserAnalyzedStatement analysis) {
+        FutureActionListener<WriteUserResponse, Long> listener = new FutureActionListener<>(r -> 1L);
+        transportCreateUserAction.execute(new CreateUserRequest(analysis.userName()), listener);
+        return listener;
+    }
+
+    @Override
+    public CompletableFuture<Long> dropUser(DropUserAnalyzedStatement analysis) {
+        FutureActionListener<WriteUserResponse, Long> listener = new FutureActionListener<>(WriteUserResponse::affectedRows);
+        transportDropUserAction.execute(new DropUserRequest(analysis.userName(), analysis.ifExists()), listener);
+        return listener;
+    }
+
+    public Iterable<User> userGetter() {
+        return users;
+    }
+
+    @Override
+    public void checkPermission(AnalyzedStatement analyzedStatement,
+                                   SessionContext sessionContext) {
+        permissionVisitor.process(analyzedStatement, sessionContext);
+    }
+
+    @Override
+    public boolean userExists(String userName) {
+        return findUser(userName) != null;
+    }
+
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        if (!event.metaDataChanged()) {
+            return;
+        }
+        users = getUsersFromMetaData(event.state().metaData().custom(UsersMetaData.TYPE));
+    }
+
+
+    @Nullable
+    User findUser(@Nullable String userName) {
+        if (userName == null) {
+            return null;
+        }
+        for (User user: userGetter()) {
+            if (userName.equals(user.name())) {
+                return user;
+            }
+        }
+        return null;
+    }
+
+    private boolean isSuperUser(String userName){
+        User user = findUser(userName);
+        return user != null && user.superuser();
+    }
+
+    private class PermissionVisitor extends AnalyzedStatementVisitor<SessionContext, Boolean> {
+
+        private final SelectStatementPermissionVisitor selectVisitor = new SelectStatementPermissionVisitor();
+
+        @Override
+        protected Boolean visitAnalyzedStatement(AnalyzedStatement analyzedStatement,
+                                                 SessionContext sessionContext) {
+            return true;
```

Comment:
I think @mfussenegger would prefer to deny all by default, so if you add new statements you are forced to implement the visit method for it. I see that point also, but don't have a strong opinion